great_expectations/experimental/datasources/interfaces.py

```diff
@@ -65,21 +75,103 @@ def datasource(self) -> Datasource:
         return self._datasource
 
     # TODO (kilo): remove setter and add custom init for DataAsset to inject datasource in constructor??
+    # This setter is non-functional: https://github.com/pydantic/pydantic/issues/3395
+    # There is some related discussion linked from that ticket which may be a workaround.
     @datasource.setter
     def datasource(self, ds: Datasource):
         assert isinstance(ds, Datasource)
         self._datasource = ds
 
-    def get_batch_request(self, options: Optional[BatchRequestOptions]) -> BatchRequest:
+    # Abstract Methods
+    def batch_request_options_template(
+        self,
+    ) -> BatchRequestOptions:
+        """A BatchRequestOptions template for get_batch_request.
+
+        Returns:
+            A BatchRequestOptions dictionary with the correct shape that get_batch_request
+            will understand. All the option values are defaulted to None.
+        """
+        raise NotImplementedError
+
+    def get_batch_list_from_batch_request(
+        self, batch_request: BatchRequest
+    ) -> List[Batch]:
         raise NotImplementedError
 
+    # End Abstract Methods
 
+    def get_batch_request(
+        self, options: Optional[BatchRequestOptions] = None
+    ) -> BatchRequest:
+        """A batch request that can be used to obtain batches for this DataAsset.
+
+        Args:
+            options: A dict that can be used to limit the number of batches returned from the asset.
+                The dict structure depends on the asset type. A template of the dict can be obtained by
+                calling batch_request_options_template.
+
+        Returns:
+            A BatchRequest object that can be used to obtain a batch list from a Datasource by calling the
+            get_batch_list_from_batch_request method.
+        """
+        if options is not None and not self._valid_batch_request_options(options):
+            raise BatchRequestError(
+                "Batch request options should have a subset of keys:\n"
+                f"{list(self.batch_request_options_template().keys())}\n"
+                f"but actually has the form:\n{pf(options)}\n"
+            )
+        return BatchRequest(
+            datasource_name=self._datasource.name,
+            data_asset_name=self.name,
+            options=options or {},
+        )
+
+    def _valid_batch_request_options(self, options: BatchRequestOptions) -> bool:
+        return set(options.keys()).issubset(
+            set(self.batch_request_options_template().keys())
+        )
+
+    def _validate_batch_request(self, batch_request: BatchRequest) -> None:
```

Comment:
Can we use a [root validator](https://docs.pydantic.dev/usage/validators/#root-validators) here?